ETT-Backend.Tests/Services/EmployeeServiceTest.cs

```diff
@@ -56,29 +97,59 @@ public void RetrieveEmployeeHoursInvalidEmail()
     }
 
     [Fact]
-    public void RetrieveEmployeeRoleString()
+    public void RetrieveAllEmployeeMetricsSuccess()
     {
-      var expectedQuery = "SELECT first_name, last_name, role FROM public.ett_employee WHERE employee_email = 'test@callibrity.com'";
-      var query = QueryGenerator.GetEmployeeInfo("test@callibrity.com");
-      Assert.Equal(expectedQuery, query);
+      System.DateTime dt = new System.DateTime();
+      List<EmployeeMetrics> dbRetVal = new List<EmployeeMetrics>()
+      {
+        new EmployeeMetrics("lol1", "lol2", "lol3", 1.0, 2.0, 3.0, 4.0, 5.0, dt),
+        new EmployeeMetrics("lul", "lul", "lul", 2.0, 2.0, 2.0, 2.0, 2.0, dt)
+      };
+
+      MockServiceProvider serviceProvider = new MockServiceProvider()
+        .AddMockDBConnection();
+
+      serviceProvider.MockDBConnection
+         .Setup(_ => _.ExecuteQuery<EmployeeMetrics>(QueryGenerator.GetEmployeeMetricsHeader() + QueryGenerator.WhereRole("Developer")))
+         .Returns(dbRetVal);
+
+      EmployeeService employeeService = new EmployeeService(serviceProvider.Build());
+      List<EmployeeResponse> response = employeeService.RetrieveAllEmployeeMetrics();
+
+      Assert.Equal("lol1", response[0].EmployeeId);
+      Assert.Equal(3.0, response[0].Billable.CurrentHours);
+      Assert.Equal(2.0, response[0].Billable.CurrentTarget);
+      Assert.Equal(1.0, response[0].Billable.TotalTarget);
+      Assert.Equal(-1.0, response[0].Growth.HoursRemaining);
+      Assert.Equal(5.0, response[0].Growth.HoursUsed);
+      Assert.Equal(4.0, response[0].Growth.TotalGrowth);
+
+      Assert.Equal("lul", response[1].EmployeeId);
+      Assert.Equal(2.0, response[1].Billable.CurrentHours);
+      Assert.Equal(2.0, response[1].Billable.CurrentTarget);
+      Assert.Equal(2.0, response[1].Billable.TotalTarget);
+      Assert.Equal(0, response[1].Growth.HoursRemaining);
+      Assert.Equal(2.0, response[1].Growth.HoursUsed);
+      Assert.Equal(2.0, response[1].Growth.TotalGrowth);
```

Comment:
Was there a reason to verify two db entries? Just curious

Reply:
the method was to retrieve all employees' metrics so I wanted to return more than one